### PR TITLE
[CSM-59] Import/export from/to file

### PR DESCRIPTION
**Pull request by flyingleafe**
Basic importing and exporting of wallets with metadata. Tested via bridge CLI.

### CODE REVIEW

File: src/Pos/Wallet/Web/Server/Methods.hs [hunk 1]
```diff
@@ -1034,6 +1042,45 @@ testResetAll | isDevelopment = deleteAllKeys >> testReset
         keyNum <- length <$> getSecretKeys
         replicateM_ keyNum $ deleteSecretKey 0
 
+----------------------------------------------------------------------------
+-- JSON backup methods
+----------------------------------------------------------------------------
+
+restoreWalletFromBackup :: WalletWebMode m => WalletBackup -> m CWallet
+restoreWalletFromBackup WalletBackup {..} = do
+    let wId = addressToCId . makePubKeyAddress $ encToPublic wbSecretKey
```
Comment (Martoon-00) [inline src/Pos/Wallet/Web/Server/Methods.hs]: There is `encToCId`

File: src/Pos/Wallet/Web/Server/Methods.hs [hunk 2]
```diff
@@ -1034,6 +1042,45 @@ testResetAll | isDevelopment = deleteAllKeys >> testReset
...
+
+    addSecretKey wbSecretKey
+    for_ accList $ \(idx, meta) -> do
+        let aIdx = fromInteger $ fromIntegral idx
+            seedGen = DeterminedSeed aIdx
+        accId <- genUniqueAccountId seedGen wId
+        createAccount accId meta
+
+    selectAccountsFromUtxoLock @WalletSscType [wbSecretKey]
+    createWalletSafe wId wMeta
```
Comment (Martoon-00) [inline src/Pos/Wallet/Web/Server/Methods.hs]: It seems safer to create wallet before accounts creation and tracking enabling

File: src/Pos/Aeson/WalletBackup.hs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        , "walletMeta" .= wMeta
+        , "accounts" .= encodeAccMap wAccounts
+        ]
+      where
+        EncryptedSecretKey prvKey passPhraseHash = skey
+        encodeAccMap accMap =
+            Array . V.fromList . map accEntry $
+            HM.toList accMap
+        accEntry (idx, accMeta) =
+            let (Object obj) = toJSON accMeta
```
Comment (Martoon-00) [inline src/Pos/Aeson/WalletBackup.hs]: Potentially unsafe coercion :unamused:

File: src/Pos/Aeson/WalletBackup.hs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+assuranceToStr :: CWalletAssurance -> Text
+assuranceToStr CWANormal = "normal"
+assuranceToStr CWAStrict = "strict"
+
+checkIfCurrentVersion :: MonadFail m => Text -> m ()
+checkIfCurrentVersion version
+    | version == currentBackupFormatVersion = pure ()
+    | otherwise =
+          fail . toString $
+          sformat ("Unsupported backup format version "%stext%", expected "%stext)
```
Comment (Martoon-00) [inline src/Pos/Aeson/WalletBackup.hs]: `toString . sformat = formatToString`

Comment (Martoon-00) [inline src/Pos/Aeson/WalletBackup.hs]: However it doesn't seem shorter ))

File: src/Pos/Wallet/Web/Server/Methods.hs (hunk 2 above)
Comment (flyingleafe) [inline src/Pos/Wallet/Web/Server/Methods.hs]: I did it this way to return wallet with correct balance without needing to get it again after all. Accounts, afaiu, does not depend on presence of wallet metadata in wallet-db, so it doesn't seem unsafe to me in some sense (maybe I'm missing something, some concurrency issues?). Anyway, it worked when I tested it.

**Comment by akegalj** [comment]
Tested by @flyingleafe in dev mode

**Pull request merged**